### ISSUE TITLE
cuda memory allocated doesn't match what I have declared

### ISSUE THREAD

**Issue by harryhan618** [pytorch/pytorch#74443]
### 🐛 Describe the bug

I want to know how memory changes during training. So I write the following code:
```python
model = nn.Sequential(nn.Linear(1024, 1024, bias=False), nn.Linear(1024, 1024, bias=False)).cuda()
x = torch.rand(4, 256, 1024).cuda()
y = model(x).sum()
memory1 = torch.cuda.memory_allocated()
y.backward()
memory2 = torch.cuda.memory_allocated()
```

so `memory1` should be 4 \* (1024\*1024\*2 + 4\*256\*1024 + 4\*256\*1024 + 128)=16777728 bytes respectively for  parameter, input, activation and sum. (I know pytorch will allocation a page of 512 bytes to hold a single float number).
And `memory2` should be 4 * (1024\*1024\*2 + 4\*256\*1024 + 128 + 1024\*1024\*2)=20972032 bytes respectively for  parameter, input, sum and gradient (the activation is freed).
And running result is what I expected: `memory1: 16777728 memory2: 20972032`

But when I change the x into `x = torch.rand(4, 480, 1024).cuda()` (or other shape which is not the power of 2), the result doesn't match what I analyze:
so `memory1` should be 4 \* (1024\*1024\*2 + 4\*480\*1024 + 4\*480\*1024 + 128)=24117760 bytes respectively for  parameter, input, activation and sum. (I know pytorch will allocation a page of 512 bytes to hold a single float number).
And `memory2` should be 4 \* (1024\*1024\*2 + 4\*480\*1024 + 128 + 1024\*1024\*2)=24642048 bytes respectively for  parameter, input, sum and gradient (the activation is freed).
But the result is `memory1: 24117760 memory2: 26214912`. The `memory1` matches but `memory2` doesn't.

I wonder what's under the hood during backward so that the memory isn't allocated as what I expect.

### Versions

pytorch 1.9,1

cc @ngimel

**Comment by harryhan618** [comment]
@ngimel
Excuse me, I just want to ask if pytorch will allocate more memory if I declare a tensor shape that is not the power of 2? (e.g. `torch.rand(4, 480, 1024, device='cuda')`)

**Comment by ngimel** [comment]
Pytorch can round some allocations up. Please use torch.discuss.org for questions, github is for bug reports.
